tap_bing_ads/fetch_ad_accounts.py

```diff
@@ -65,4 +65,17 @@ def request_customer_id(access_token: str, developer_token: str) -> int:
         obj=response,
     )
 
+    # The request return 200 even if we fail to get customer_id
+    error = get_field(
+        "s:Envelope",
+        "s:Body",
+        "s:Fault",
+        "detail",
+        "AdApiFaultDetail",
+        "Errors",
+        obj=response,
+    )
+
+    if error:
+        raise Exception(json.dumps(error))
```

Comment:
To determine whether the issue is with credentials, I think we need to check the `AdApiError/Code` value.

From the [documentation](https://learn.microsoft.com/en-us/advertising/guides/handle-service-errors-exceptions?view=bingads-13#code-105), error codes `105` and `107` seems to map to invalid credentials, whereas the `117` is due to rate limiting.

[There are also a bunch of other possible error codes](https://learn.microsoft.com/en-us/advertising/guides/operation-error-codes?view=bingads-13), no clue which can be returned here.

The approach we've taken in most other taps (that I know of) is to do a lazy `raise InvalidCredentialsException()` and then `os.exit(5)` somewhere up top to trigger the automatic revocation logic.